autoendpoint/src/routers/fcm/error.rs

```diff
@@ -86,11 +86,8 @@ impl ReportableError for FcmError {
 
     fn metric_label(&self) -> Option<&'static str> {
         match &self {
-            FcmError::InvalidAppId(_) | FcmError::NoAppId => {
-                Some("notification.bridge.error.fcm.badappid")
-            }
-            FcmError::Upstream { status, .. } if status == "RESOURCE_EXHAUSTED" => {
-                Some("notification.bridge.error.fcm.resource_exhausted")
+            FcmError::InvalidAppId(_) | FcmError::NoAppId | FcmError::Upstream { .. } => {
```

Comment:
I don't think we want `Upstream` here with `handle_error` already emitting a metric for it: as we'd get a metric emitted twice for it.

I think all `FcmErrors` will be passed through `handle_error` (or going to sentry, which `InvalidAppId`/`NoAppId` are set to do)? If so I'd argue we shouldn't return any metric labels here at all (for now)

Reply:
Ah, you're right, I forgot that `handle_error()` records the metric and then returns the original error for later processing.
That said, we do have some duplication already. I'll add a comment so I can remember that later.
I note that `RouterError::TooMuchData(_)` produces `notification.bridge.error {too_much_data}` [🔗](https://github.com/mozilla-services/autopush-rs/blob/559b6537eb8b70bed4dd702403af8cc0dde4f57b/autoendpoint/src/routers/common.rs#L120-L121)  and `notification.bridge.error.too_much_data` [🔗](https://github.com/mozilla-services/autopush-rs/blob/559b6537eb8b70bed4dd702403af8cc0dde4f57b/autoendpoint/src/routers/mod.rs#L195-L196) Should I remove that as well so that we only have one? (I can add that as a different PR.)
I'll also note that we might want to have a `ReportableError::metric_tags()` function similar to how we have `extras()` to provide the additional information, since that seems to be a pattern we're hitting.

Reply:
I'm fine with leaving it for now. 

We already have `ReportableError::tags` (it defaults to returning nothing, so you just have to implement it to override)